File: recursion_func/hanoi_tower.c
// Hanoi Tower Algorithm

#include <stdio.h>

void hanoi_tower(int n, char from, char temp, char to) {

	if (n == 1) {
		printf("No.1 disk moves from %c to %c\n", from, to);
	} else {
		hanoi_tower(n - 1, from, to, temp);
		printf("No.%d disk moves from %c to %c\n", n, from, to);
		hanoi_tower(n - 1, temp, from, to);
	}
}

int main() {
	hanoi_tower(3, 'A', 'B', 'C');
	return 0;
}

File: recursion_func/hanoi_tower.py
# Hanoi Tower Algorithm

def hanoi_tower(n, start, temp, to):
    if n == 1:
        print("No. 1 disk moves from", start, "to", to)
    else:
        hanoi_tower(n - 1, start, to, temp)
        print("No.", n, "disk moves from", start, "to", to)
        hanoi_tower(n - 1, temp, start, to)

hanoi_tower(3, "A", "B", "C")
